app/leagues/page.tsx

```diff
@@ -3,52 +3,47 @@
 
 import React, { JSX } from 'react';
 import { LeagueCard } from '@/components/LeagueCard/LeagueCard';
+import { IUser } from '@/api/apiFunctions.interface';
+import { getCurrentUser } from '@/api/apiFunctions';
+import { getUserLeagues } from '@/utils/utils';
+
+interface Props {
+  searchParams?: { id: IUser['id'] };
+}
 
 /**
  * Renders the leagues component.
- * @returns {JSX.Element} The rendered leagues component.
+ * @param {Props} props The props for the leagues component.
+ * @returns {JSX.Element | undefined} The rendered leagues component.
  */
-const Leagues = (): JSX.Element => {
+const Leagues = async ({
+  searchParams,
+}: Props): Promise<JSX.Element | undefined> => {
```

Comment:
How is this returning undefined? It always returns JSX.

Reply:
because I have an error check:

```js
  if (!userId || userId === '') {
    return;
  }
 ```

This returns undefined until the userId is loaded in state

Reply:
agreed.  This will be fixed when I change the dynamic route structure to what is in my presentation.

Reply:
@shashilo am I still changing the route for this or do you have that with the current dynamic routes you sent me the screenshot of?